### ISSUE TITLE
How to compile and run the samples?

### ISSUE THREAD

**Issue by MoaLaiSkirulais** [vurtun/nuklear#127]
Hi all
I am trying on ubuntu linux. The allegro sample.
And get this:

``` bash
root@easatmidev:/var/www/html/xtras/users/MoaLaiSkirulais/nuklear/demo/allegro5# make
rm -f bin/demo
cc main.c -std=c89 -pedantic -o bin/demo -lallegro -lallegro_primitives -lallegro_main -lGL -lm -lGLU -lGLEW
In file included from main.c:29:0:
nuklear_allegro.h:16:30: fatal error: allegro5/allegro.h: No such file or directory
 #include <allegro5/allegro.h>
                              ^
compilation terminated.
make: *** [demo] Error 1
```

May be a newby questions i know.
Congrats for the work i wish to use for an own project, an app client for rest api.
thnx

**Comment by fr500** [comment]
This isn't really a nuklear issue..
It means you don't have allegro installed or in your include path

Not on ubuntu atm but 
sudo apt-get install allegro5 allegro5-dev or sudo apt-get install liballegro5 liballegro5-dev should do the trick

**Comment by MoaLaiSkirulais** [comment]
Thnx
Let me explain, reading the introduction i understood there are no "prerequisites"
I also tried the example and get a simmilar error

````bash
root@easatmidev:/var/www/html/xtras/users/MoaLaiSkirulais/nuklear# cd example/
root@easatmidev:/var/www/html/xtras/users/MoaLaiSkirulais/nuklear/example# make
rm -f bin/file_browser
cc -std=c99 -pedantic -O2 -o bin/file_browser file_browser.c -lglfw -lGL -lm -lGLU -lGLEW
file_browser.c:15:21: fatal error: GL/glew.h: No such file or directory
 #include <GL/glew.h>
                     ^
compilation terminated.
make: *** [file_browser] Error 1
````
The question is: is possible to run some kind of Nulear "Hello World" sample in an ubuntu server without any desktop enivornment?

**Comment by fr500** [comment]
There are no prerequisites for developers, you can implement your own backend and you're not forced to use allegro or GL anything like that. You can use whatever you want granted you have the knowledge to implement the backend.

The samples/examples have some prerequisites of course and I don't think there's a DRM/KMS GL sample that you can possibly run without a DE.

As for that error, you have to install glew like in the other case you had to install allegro.

**Comment by dumblob** [comment]
Hi @MoaLaiSkirulais, there is an effort https://github.com/vurtun/nuklear/issues/101 going on which should provide what you are asking for.

**Comment by vurtun** [comment]
@MoaLaiSkirulais If you are on Ubuntu or any Linux with `X11` you can try out those demos (`X11`. `X11_opengl2`, `X11_opengl3`). They don't have any external library dependencies outside xlib (platform dependency needed to create a window, handle input and draw to screen, ...) and OpenGL for both OpenGL demos.

**Comment by nemuos** [comment]
I followed these steps (building glfw3 from source and compiling the example):
1. cd to glfw3 
2. sudo cmake -G "Unix Makefiles" -DCMAKE_INSTALL_PREFIX:PATH=/usr
3. sudo make -j4
4. sudo make install
5. cd to nuklear/examples
6. cc -std=c99 -pedantic -O2 -o bin/file_browser file_browser.c -lglfw -lGL -lm -lGLU -lGLEW -I/usr/include -L/usr/lib

All I am getting is the errors
------------------------------------------------------------------------------------
/tmp/cchozgb6.o: In function `scroll_input':
file_browser.c:(.text+0x1b25b): undefined reference to `glfwGetWindowUserPointer'
/tmp/cchozgb6.o: In function `text_input':
file_browser.c:(.text+0x1b3d4): undefined reference to `glfwGetWindowUserPointer'
/tmp/cchozgb6.o: In function `main':
file_browser.c:(.text.startup+0x5b): undefined reference to `glfwSetErrorCallback'
file_browser.c:(.text.startup+0x77): undefined reference to `glfwWindowHint'
file_browser.c:(.text.startup+0x86): undefined reference to `glfwWindowHint'
file_browser.c:(.text.startup+0x95): undefined reference to `glfwWindowHint'
file_browser.c:(.text.startup+0xae): undefined reference to `glfwCreateWindow'
.
.
-------------------------------------------------------------------------------------

Can anyone help me solve the problem.
Oh, well I added add_defintions(-fPIC) in CMakeLists.txt file as suggested by some solutions to this problem, but it didn't work too.
